File: src/main/java/br/anhembi/funmodechild/repository/CustomerRepository.java
package br.anhembi.funmodechild.repository;

import br.anhembi.funmodechild.entity.Customer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface CustomerRepository extends MongoRepository<Customer, String> {

    Customer findByEmail(String email);
}

File: src/main/kotlin/br/anhembi/funmodechild/repository/CustomerRepository.kt
package br.anhembi.funmodechild.repository

import br.anhembi.funmodechild.entity.Customer
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface CustomerRepository : MongoRepository<Customer, String> {
    fun findByEmail(email: String): Customer?
}
